packages/providers/src.ts/base-provider.ts

```diff
@@ -1257,20 +1257,26 @@ export class BaseProvider extends Provider implements EnsProvider {
 
     async getCode(addressOrName: string | Promise<string>, blockTag?: BlockTag | Promise<BlockTag>): Promise<string> {
         await this.getNetwork();
-        const params = await resolveProperties({
-            address: this._getAddress(addressOrName),
-            blockTag: this._getBlockTag(blockTag)
-        });
-
-        const result = await this.perform("getCode", params);
+        addressOrName = await addressOrName;
+        const { shard, realm, num } = getAccountFromAddress(addressOrName);
+        const shardNum = BigNumber.from(shard).toNumber();
+        const realmNum = BigNumber.from(realm).toNumber();
+        const accountNum = BigNumber.from(num).toNumber();
+        const endpoint = '/api/v1/contracts/' + shardNum + '.' + realmNum + '.' + accountNum;       
```

Comment:
Instead of constructing the string version of the account every time we could change the `Account` type to define a concrete `class` and implement a `toString` method so that we can do `getAccountFromAddress(address).toString()` and use that

Reply:
We could resolve this in a separate PR

Reply:
I agree, it would be cleaner to implement a `toString` method